pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -1246,7 +1245,9 @@ func GetPodAntiAffinityTerms(podAntiAffinity *v1.PodAntiAffinity) (terms []v1.Po
 	return terms
 }
 
-func getMatchingTopologyPairs(pod *v1.Pod, nodeInfoMap map[string]*schedulercache.NodeInfo) (*topologyPairsMaps, error) {
+// getMatchingAntiAffinityTopologyPairs is an initialization helper to calculate on each existing pod
+// of each node, whether (1) it has PodAntiAffinity and (2) ALL AffinityTerms matches incoming pod
+func getMatchingAntiAffinityTopologyPairs(pod *v1.Pod, nodeInfoMap map[string]*schedulercache.NodeInfo) (*topologyPairsMaps, error) {
```

Comment:
Am I misreading, or do two funcs have the same name here?

Reply:
Yes, it does...

- func getMatchingAntiAffinityTopologyPairs(pod *v1.Pod, nodeInfoMap map[string]*schedulercache.NodeInfo) (*topologyPairsMaps, error)
It's used by `metadata.go` to calculate an overall topologyPairs map on all nodes. Underneath, it invokes `getMatchingAntiAffinityTopologyPairsOfPod()`
- func (c *PodAffinityChecker) getMatchingAntiAffinityTopologyPairs(pod *v1.Pod, allPods []*v1.Pod) (*topologyPairsMaps, error)
It's a method receiver of `*PodAffinityChecker`. Which actually also invokes `getMatchingAntiAffinityTopologyPairsOfPod()`.

Reply:
I missed the fact that one of them is a receiver. I am finding it difficult to follow the logic in this file.

These functions are doing different things, so I don't think we should give them the same name. Since we calculate/compute the topology pairs for metadata, then do a lookup during predicate execution, maybe we should name one "calculate" and one "get"?

Reply:
Maybe we coulde rename like this:

- getMatchingAntiAffinityTopologyPairs -> initMatchingAntiAffinityTopologyPairs - as it's only invoked once in `metadata.go`
- (c *PodAffinityChecker) getMatchingAntiAffinityTopologyPairs -> (c *PodAffinityChecker) calculateMatchingAntiAffinityTopologyPairs
- getMatchingAntiAffinityTopologyPairsOfPod() - keep it as is